.buildkite/pipeline.yml

```diff
@@ -32,6 +33,28 @@ steps:
       - "dist/*.tar.gz"
       - "logs/**/*"
 
+  ######################
+  # Publish Android Artifacts
+  ######################
+  - label: "🧪 Build and upload Android Library to S3"
+    command: .buildkite/publish-android.sh
+    depends_on:
+      - "bundle-js"
+    plugins:
+      - docker#v3.8.0:
+          # Use a temporary Docker container that has both Android & Node setup
+          # We're looking to remove the Docker usage as soon as possible.
+          image: "public.ecr.aws/automattic/android-build-image:bc7c60a898d89d968aa8202c6b5bd08f91c1779e"
+          propagate-environment: true
+          environment:
+            # DO NOT MANUALLY SET THESE VALUES!
+            # They are passed from the Buildkite agent to the Docker container
+            - "AWS_ACCESS_KEY"
+            - "AWS_SECRET_KEY"
+    agents:
+      queue: "android"
+    if: "build.tag != null || build.pull_request.id != null || build.branch == 'trunk'"
```

Comment:
Unfortunately this won't work if the branch is already built when the PR is opened. So, either we'll have to rebuild when the PR is opened, or we need to address this in a different way.

A possible solution might be to remove this condition and skip the actual publishing in `.buildkite/publish-android.sh` script. The main problem with that is the android publishing job would look like it was successful when in fact it didn't publish anything. Having some logs explaining this might be helpful, but still doesn't feel like a great solution to me.

My preferred solution would be to have this conditional at the pipeline level, similar to our other Android projects, but I don't know how that'd affect the iOS stack.

I am also open to any other suggestions. However, I want to note that publishing each commit is not currently supported by `publish-to-s3-gradle-plugin` and even if we decide to add it, I don't think it'd be a good solution especially since this project requires us to publish `react-native` dependencies alongside our library.

Reply:
Hi @oguzkocer , thanks for explanation here. I have a question: does this also affect building PR for a tag? I'm trying to figurer out why the library is not published for a new tag.

Reply:
@wzieba I don't think this one would affect a new tag, because after a tag is created an entirely new build should be triggered. We have had occasional GitHub issues in the past where GitHub failed to notify Buildkite. Which tag are you trying to build? Maybe I can take a look at it.

> My preferred solution would be to have this conditional at the pipeline level, similar to our other Android projects, but I don't know how that'd affect the iOS stack.

We discussed this with @jkmassel on Slack and he agreed to this solution. I'll try to open a PR for that now.

Reply:
@wzieba I just started working on that ^ suggestion and realized the builds are turned off for tags. I'll address both issues together.

Reply:
Gotcha, thank you!

Reply:
@wzieba The issue with the tags should now be resolved.

Reply:
To close the loop, this conditional is no longer necessary and removed in https://github.com/woocommerce/WooCommerce-Shared/pull/81.